src/Guide/Api/Guider.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+{- | 'Guider' monad with 'Config' to replace servant's 'Handler'. -}
+module Guide.Api.Guider
+       ( Guider (..)
+       , GuiderServer
+       , guiderToHandler
+       ) where
+
+import Imports
+
+import Servant (Handler (..), ServantErr)
+import Servant.Server.Generic
+
+import Guide.Config (Config (..), def)
+
+
+-- | Custom 'Guider' type holds the 'Config' always on hand.
+newtype Guider a = Guider
+  { runGuider :: ReaderT Config IO a
+  } deriving (Functor, Applicative, Monad, MonadIO, MonadReader Config)
+
+instance MonadError ServantErr Guider where
+  throwError :: ServantErr -> Guider a
+  throwError = liftIO . throwIO
+
+  catchError :: Guider a -> (ServantErr -> Guider a) -> Guider a
+  catchError (Guider m) f =
+    liftIO $ (runReaderT m def) `catch` (\e -> runReaderT (runGuider (f e)) def)
```

Comment:
"If it typechecks, it works" doesn't work here 🙂 

You can't ignore the reader argument and run the handler with `def` as the config. It would be buggy behavior.

Here's something that should work (I haven't tried it):

```haskell
  catchError (Guider m) f = Guider $ ReaderT $ \config ->
    runReaderT m config `catch` (\err -> runReaderT (runGuider (f err)) config)
```